src/AppProviders.tsx

```diff
@@ -33,15 +34,17 @@ export function AppProviders({ children }: ProvidersProps) {
               <BrowserRouterProvider>
                 <I18nProvider>
                   <WalletProvider>
-                    <ModalProvider>
-                      <PortfolioProvider>
-                        <MarketDataProvider>
-                          <TransactionsProvider>
-                            <DefiManagerProvider>{children}</DefiManagerProvider>
-                          </TransactionsProvider>
-                        </MarketDataProvider>
-                      </PortfolioProvider>
-                    </ModalProvider>
+                    <KeepKeyProvider>
```

Comment:
Another provider?

Reply:
I've gone back and forth on this a few times in my head.

I've landed on `KeepKeyProvider` containing too much hardware wallet-specific logic to muddy `WalletProvider` with. I expect if we add first-class support for another wallet we'll refactor `KeepKeyProvider` into a `HardwareWallet` provider.